File: ruru/src/orm/note.rb
 # note model - ch06 section 2
class Note < Tina4::ORM
  self.field_mapping = {
    "category" => "tag"
  }

  integer_field :id, primary_key: true, auto_increment: true
  string_field :title, nullable: false
  string_field :content, default: ""
  string_field :category, default: "general"
  integer_field :pinned, default: 0
  datetime_field :created_at
  datetime_field :updated_at
end

File: documentation-testing/pypy/src/orm/note.py
# ch06 section 2 — "Defining a Model"
# VERBATIM from the chapter. Model definition exactly as documented.
from tina4_python.orm import ORM, IntegerField, StringField, BooleanField, DateTimeField

class Note(ORM):
    table_name = "notes"

    id = IntegerField(primary_key=True, auto_increment=True)
    title = StringField(required=True, max_length=200)
    content = StringField(default="")
    category = StringField(default="general")
    pinned = BooleanField(default=False)
    created_at = DateTimeField()
    updated_at = DateTimeField()
